newshomepages/html.py

```diff
@@ -0,0 +1,113 @@
+import typing
+from pathlib import Path
+
+import click
+from playwright.sync_api import sync_playwright
+from retry import retry
+from rich import print
+
+from . import utils
+
+
+def _read_script_from_file(filename: str):
+    """Read and execute Javascript code from a file to a `Page` object."""
+    with open(filename) as f:
+        return f.read()
+
+
+@click.command()
+@click.argument("handle")
+@click.option("-o", "--output-dir", "output_dir", default="./")
+@click.option("-w", "--wait", "wait", default=5000)
+def cli(
+    handle: str,
+    output_dir: str = "./",
+    wait: str = "5000",
+):
+    """Save HTML for the provided homepage."""
+    site = utils.get_site(handle)
+    output_path = Path(output_dir)
+    _save_html(site, output_path, wait=int(wait))
+
+
+@retry(tries=3, delay=5, backoff=2)
+def _save_html(
+    site: typing.Dict,
+    output_path: Path,
+    wait: int = 5000,
+):
+    """Shoot the provided site."""
+    print(f":camera: Fetching HML for {site['name']}")
+
+    # Set the output path
+    output_path.mkdir(parents=True, exist_ok=True)
+
+    with sync_playwright() as playwright:
+        context = utils._load_persistent_context(playwright)
+        page = utils._load_new_page_disable_javascript(
+            context=context,
+            url=site["url"],
+            wait_seconds=int((site["wait"] or wait) / 1000),
+            handle=site["handle"],
+            full_page=True,
+        )
+
+        single_file_pre_load_extensions = [
+            str(
+                utils.EXTENSIONS_PATH
+                / "singlefile"
+                / "javascript"
+                / "single-file-bootstrap.js"
+            ),
+            str(
+                utils.EXTENSIONS_PATH
+                / "singlefile"
+                / "javascript"
+                / "single-file-hooks-frames.js"
+            ),
+            str(
+                utils.EXTENSIONS_PATH
+                / "singlefile"
+                / "javascript"
+                / "single-file-frames.js"
+            ),
+        ]
+        for f in single_file_pre_load_extensions:
+            page.evaluate(_read_script_from_file(f))
+
+        post_load_script = str(
+            utils.EXTENSIONS_PATH / "singlefile" / "javascript" / "single-file.js"
+        )
```

Comment:
All the files you're reading appear to be in the same subdirectory? If we're going to have this broken out function, you could stow away the directory prefix stuff up there so you don't have to repeat it down here.